### ISSUE TITLE
Documentation for newly implemented game concepts

### ISSUE THREAD

**Issue by LynxAbraxas** [civctp2/civctp2#74]
In relation to #49 (https://github.com/civctp2/civctp2/issues/49#issuecomment-457877924):

Do we have some place where newly implemented game concepts are documented? 
If not, additions from mods should be documented in a kind of manual to not loos track of what has been done and to avoid that the code is the only "documentation".

**Comment by MartinGuehmann** [comment]
I don't think so. And what do you call a mod. I would restrict it to modifying game files, but not source files. We added a few things from classical mods, that are most prominently CityMod and GoodMod. We did not add things from the slic files, since the idea was that this stuff should be implemented in the source.

However, there are still a few things we could include, I thing the stuff from map.txt concerning world generation would be a good addition.

Other things are bit questionable, the people tweaked the priorities in the goals for attack and siege hoping that they would improve the AI. However, that does not help if the logic in the code is flawed.

Beside that #49 addresses something changed by Activision, so if it all then it would be in the official manual if at all.

**Comment by LynxAbraxas** [comment]
Hm, then what is this EMOD that pops up in the code?
Improvements to the AI are not necessarily needed by a gamer, but new options/behaviours like #73 would do well.

> Beside that #49 addresses something changed by Activision, so if it all then it would be in the official manual if at all.

Agree, but if activision missed it, our documentation for new stuff could also include a section like "explanations to activision manual and things that are missing in the original manual"

**Comment by MartinGuehmann** [comment]
> Hm, then what is this EMOD that pops up in the code?

That are modifciations by E (his user name on Apolyton, ekmek on CivFanatics). So he added quite some stuff to the game. However, he is not a programmer, so the things are half done. And I eded up to fix the things or care that they are not in the way.

> Improvements to the AI are not necessarily needed by a gamer, but new options/behaviours like #73 would do well.

That depends, originally one of the big complains about the game was the weak AI, and that was one of our issues:

Fix the AI

Sounds simple but a game AI is quite complex so there are many issues. It least it should be a bit challenging, an AI that you can beat at every wonder is a bit dull. In principle, every game has an AI that is flawed in one respect or in another. Simply because AIs are complex, so you have a lot of chances to screw up.

Of course, a game that runs faster and needs less memory is useful for the gamer, especially if it does not crash because it runs out of memory.

But you are right, we need also a few more goodies for the next release.
 
> Agree, but if activision missed it, our documentation for new stuff could also include a section like "explanations to activision manual and things that are missing in the original manual"

True.
